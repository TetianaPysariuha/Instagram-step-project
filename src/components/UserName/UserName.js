import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './UserName.module.scss';
import defaultAvatar from './default-avatar.jpg';

function UserName({
  image, nickname, additionalString, comment, additionalStringFunc,
}) {
  return (
    <div className={styles.userNameContainer}>
      <Link to={`/${nickname}/`} className={styles.avatarLink}>
        <img src={image} alt="avatar" className={styles.avatar} />
      </Link>
      <div className={styles.nicknameWrraper}>
        <div>
          <Link to={`/${nickname}/`} className={styles.nickname}>
            <span>{nickname}</span>
          </Link>
          <p className={styles.comment}>{comment}</p>
        </div>
        {additionalString && <p className={styles.additionalString} onClick={additionalStringFunc}>{additionalString}</p>}
      </div>
    </div>
  );
}

UserName.prototype = {
  image: PropTypes.string,
  nickname: PropTypes.string.isRequired,
  additionalString: PropTypes.string,
  comment: PropTypes.string,
  additionalStringFunc: PropTypes.func,
};

UserName.defaultProps = {
  image: defaultAvatar,
  additionalString: '',
  additionalStringFunc: () => {},
};

export default UserName;
