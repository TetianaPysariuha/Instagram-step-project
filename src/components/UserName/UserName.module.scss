.userNameContainer {
    display: flex;
    gap: 10px;
    padding: 8px 16px;
    max-width: 300px;
}

.avatarLink {
    width: 32px;
    height: 32px;
}

.avatar {
    width: 32px;
    height: 32px;
    border-radius: 50%;
    border: 1px solid rgb(240, 240, 240);
}

.nicknameWrraper {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    gap: 5px;
    margin: auto 0;
}

.nickname {
    color: rgb(32, 32, 32);
    text-decoration: none;
}

.additionalString {
    color: rgb(142, 142, 142);
    margin: 0;
    font-weight: 200;
    font-size: 12px;
}

.comment {
    display: inline;
    font-weight: 500;
    margin-left: 5px;
    font-size: 14px;
}