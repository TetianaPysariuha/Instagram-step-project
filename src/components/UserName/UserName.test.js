import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import UserName from './UserName';

function Component() {
  return (
    <BrowserRouter>
      <UserName />
    </BrowserRouter>
  );
}

describe('UserName renders', () => {
  test('Should Header render', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
