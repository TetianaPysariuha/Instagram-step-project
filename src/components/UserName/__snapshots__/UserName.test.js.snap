// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header reders Should Header render 1`] = `
<DocumentFragment>
  <div
    class="userNameContainer"
  >
    <a
      class="avatarLink"
      href="/undefined/"
    >
      <img
        alt="avatar"
        class="avatar"
        src="default-avatar.jpg"
      />
    </a>
    <div
      class="nicknameWrraper"
    >
      <div>
        <a
          class="nickname"
          href="/undefined/"
        >
          <span />
        </a>
        <p
          class="comment"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
